### PR TITLE
add StartStopButton

### LINKED ISSUE #28: Create UI - Start and Stop button
## User story
1. As a user
2. I want a button to start/stop the signals
3. So that I can freeze the input and restart with current inputs from the data source.

## Acceptance criteria
- [x] Start/Start button visible clearly.
- [x] Each channel should possess its own button.
- [x] Start button starts the signals with current input from the data source.
- [x] Stop button stops the signal visualization.
- [x] Stop button should not interfere with the signal data input to the frontend from backend.

## Definition of done (DoD)
- [ ] Application compiles successfully
- [ ] Code is documented
- [ ] Feature/Bug/Change is tested by at least one unit or e2e test
- [ ] Tests have been passed without warnings (except "deprecated" warnings)
- [ ] Changes have been reviewed
- [ ] PR has been merged to dev branch
- [ ] New dependencies have been added to bill of materials
- [ ] Software architecture diagram has been updated
- [ ] All acceptance criteria are fulfilled
- [ ] Screenshot is attached to issue

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release

### PR DESCRIPTION
**Pull request by rabbit-zero**
created button for start and stop plotting the signal

### CODE REVIEW

File: Apps/frontend/src/App.svelte [hunk 1]
```diff
@@ -3,6 +3,7 @@
   import Logo from "./views/Logo.svelte";
   import OnOffButton from "./views/OnOffButton.svelte";
   import Oscilloscope from "./views/Oscilloscope.svelte";
+  import StartStopButton from "./views/StartStopButton.svelte";
```
Comment (leandertolksdorf) [inline Apps/frontend/src/App.svelte:6]: unused import

File: Apps/frontend/src/views/SineWave.svelte [hunk 2]
```diff
@@ -65,10 +74,11 @@
...
-    webGLPlot.update();
-    window.requestAnimationFrame(newFrame);
+      update();
+      webGLPlot.update();
+      window.requestAnimationFrame(newFrame);
   }
 </script>
 
 <canvas bind:this={canvasElement} />
+<StartStopButton on:startStop={async (event)=> {hasStarted = event.detail.buttonValue}} />
```
Comment (leandertolksdorf) [inline Apps/frontend/src/views/SineWave.svelte:84]: The component should be placed in the Oscilloscope.svelte component in my opinion.

**Review by leandertolksdorf — CHANGES_REQUESTED** [review]
Sorry if I missed something, but this does the same as #66, except that in #66 the drawing resumes at the point it stopped, right?

**Comment by motschel123** [comment]
I'm not certain about the requirements, but in my opinion we should discard all received stream data and start from the beginning (cleaning the old curves) as we click resume

**Review by jenswaechtler — CHANGES_REQUESTED** [review]
The button changes size when I click on it. Perhaps it depends on the text field. Not necessary scope of this issue, but perhaps you know a fix.

File: Apps/frontend/src/views/Oscilloscope.svelte [hunk 3]
```diff
@@ -17,9 +20,16 @@
...
-      waveElement.updateBuffer(samples);
+      if(hasStarted){
+        let samples = new Float64Array(message.data);
+        waveElement.updateBuffer(samples);
+
+      }else{
+        
+      }
+      
+
```
[the remark this answers is not in the document]
Comment (rabbit-zero) [reply Apps/frontend/src/views/Oscilloscope.svelte:32]: oh, I forgot to delete that

**Review by jenswaechtler — APPROVED** [review]
Looks good to me.

**Comment by PhlppKrmr** [comment]
Just to clarify:
What @rabbit-zero implemented is perfectly fine, only the naming of the issue is a little onfortunate (start and freeze would've most likely been clearer).
We also asked Mr. Eckert on Monday whether he wants the feature @motschel123 mentions (clear canvas and restart). This is really low prio cause we can simply refresh the page itself, we don't need a button for that.

File: Apps/frontend/src/views/StartStopButton.svelte [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	const handleStartStop = async () => {
+		hasStarted = !hasStarted;
+		hasStarted ? symbol = stop : symbol = play;	
+		dispatch('startStop', {buttonValue: hasStarted})
+	}
+	
+	
+</script>
+
+<button class="button-style button-up" on:click={handleStartStop}>
```
Comment (PhlppKrmr) [inline Apps/frontend/src/views/StartStopButton.svelte:20]: Funtionality works fine, only the styling could be improved (as @jenswaechtler already mentioned the text forces a different button size)
-> For me personally a simple button with the icons would be enough (as long as the size stays the same then)

**Review by PhlppKrmr — APPROVED** [review]

**Comment by jenswaechtler** [comment]
The pull request is against the main branch. This should be changed to dev.

**Review by motschel123 — COMMENTED** [review]
Should be into dev, as mentioned.
In my opinion Text/button size changes are not a problem for now.
Otherwise LGTM.

**Review by leon-juenemann — COMMENTED** [review]
I just changed the target to dev. CI/CD should pass now.

**Review by jandegen — APPROVED** [review]

**Review by jenswaechtler — APPROVED** [review]

**Pull request merged**